pkg/db/tableswapper/tableswapper.go

```diff
@@ -0,0 +1,63 @@
+package tableswapper
+
+import (
+	"fmt"
+	"time"
+
+	"gorm.io/gorm"
+
+	"github.com/openshift/sippy/pkg/db"
+)
+
+// TableSwapper is used to replace an existing table entirely, by creating a temporary staging table and renaming
+// it later.
+type TableSwapper struct {
+	dbc          *db.DB
+	tableName    string
+	oldTable     string
+	stagingTable string
+}
+
+func New(dbc *db.DB, tableName string) *TableSwapper {
+	timestamp := time.Now().UnixMilli()
+
+	return &TableSwapper{
+		dbc:          dbc,
+		tableName:    tableName,
+		stagingTable: fmt.Sprintf("%s_staging_%d", tableName, time.Now().UnixMilli()),
+		oldTable:     fmt.Sprintf("%s_old_%d", tableName, timestamp),
+	}
+}
+
+// CreateStagingTable creates the temporary staging table.  Use dbc.Table(stagingTable).Save(X) to write records
+// to the temporary staging table instead of the default.
+func (ts *TableSwapper) CreateStagingTable() (string, error) {
+	// Create a temporary staging table
+	if err := ts.dbc.DB.Exec(fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s AS TABLE %s WITH NO DATA", ts.stagingTable, ts.tableName)).Error; err != nil {
+		return "", err
+	}
+
+	return ts.stagingTable, nil
+}
+
+// Swap performs the swap by renaming the old table, and replacing it with the staging table.
+func (ts *TableSwapper) Swap() error {
+	// Swap tables and drop the old one
+	return ts.dbc.DB.Transaction(func(tx *gorm.DB) error {
+		if err := tx.Exec(fmt.Sprintf("ALTER TABLE %s RENAME TO %s", ts.tableName, ts.oldTable)).Error; err != nil {
+			return err
+		}
+		if err := tx.Exec(fmt.Sprintf("ALTER TABLE %s RENAME TO %s", ts.stagingTable, ts.tableName)).Error; err != nil {
+			return err
+		}
+		return tx.Exec(fmt.Sprintf("DROP TABLE %s", ts.oldTable)).Error
+	})
+}
+
+// Cleanup removes any leftover tables.  Good idea to defer this after calling CreateStagingTable.
+func (ts *TableSwapper) Cleanup() error {
+	if err := ts.dbc.DB.Exec(fmt.Sprintf("DROP TABLE IF EXISTS %s", ts.stagingTable)).Error; err != nil {
```

Comment:
Can't recall if these Exec statements log nicely still, I recall shutting something off, but if not we'd want logging on all major operations in this module. Disregard if this already appears in the log cleanly.